receiver.go

```diff
@@ -74,10 +75,10 @@ func newReceiveHandler(enc encoder, dec byteReadingDecoder, rmCallCh chan callRe
 		wrapErrorFunc: wef,
 	}
 	r.messageHandlers = map[MethodType]messageHandler{
-		MethodNotify:   {dispatchFunc: r.receiveNotify, messageLength: 3},
-		MethodCall:     {dispatchFunc: r.receiveCall, messageLength: 4},
+		MethodNotify:   {dispatchFunc: r.receiveNotify, messageLength: 4},
+		MethodCall:     {dispatchFunc: r.receiveCall, messageLength: 5},
 		MethodResponse: {dispatchFunc: r.receiveResponse, messageLength: 4},
-		MethodCancel:   {dispatchFunc: r.receiveCancel, messageLength: 3},
+		MethodCancel:   {dispatchFunc: r.receiveCancel, messageLength: 4},
```

Comment:
> Why does cancel need an extra field for the log tags? Aren't the log tags effectively the ones attached to the original call?

You're right, we should be able to retrieve the tags from the original context.

> Why doesn't response have log tags? I can imagine the server perhaps adding a few more log tags or something.

We never send a context back from a response, so we don't have a mechanism for sending back new tags. If this is desired, we can do it, but there wasn't a requirement for it.